api/src/main/java/org/ehcache/expiry/Duration.java

```diff
@@ -48,6 +48,40 @@ public static Duration of(long length, TimeUnit timeUnit) {
     return new Duration(length, timeUnit);
   }
 
+  public static Duration of(java.time.Duration duration) {
+    if (duration == null) {
+      return null;
+    }
+    if (duration.isNegative()) {
+      throw new IllegalArgumentException("Ehcache duration cannot be negative and so does not accept negative java.time.Duration: " + duration);
+    }
+    if (duration.isZero()) {
+      return Duration.ZERO;
+    } else {
+      long nanos = duration.getNano();
+      if (nanos == 0) {
+        return of(duration.getSeconds(), TimeUnit.SECONDS);
+      }
+      long seconds = duration.getSeconds();
```

Comment:
I think we should consider that `Duration.of(Long.MAX_VALUE, ChronoUnit.NANOS)` is the infinite and give a constant to represent it.

Reply:
Yes, the `java.time.Duration` javadocs suggests essentially that in the saying that `Duration.of(Long.MAX_VALUE, ChronoUnit.NANOS)` is greater than the current estimated age of the universe.